src/hotspot/share/opto/mulnode.cpp

```diff
@@ -1088,6 +1088,18 @@ Node* LShiftNode::IdealIL(PhaseGVN* phase, bool can_reshape, BasicType bt) {
       }
     }
   }
+  // Transform is legal, but check for profit.  Avoid breaking 'i2s'
```

Comment:
I would add a comment that explicitly states the pattern we are looking at, this makes reading the code much faster imo:
```suggestion
  // Check for "(con0 - X) << con1" 
  // Transform is legal, but check for profit.  Avoid breaking 'i2s'
```